lib/galaxy/webapps/galaxy/services/tools.py

```diff
@@ -240,7 +281,12 @@ def _handle_inputs_output_to_api_response(self, trans, tool, target_history, var
             rval["implicit_collections"].append(output_dict)
 
         trans.security.encode_all_ids(rval, recursive=True)
-        return rval
+
+        # Encoding the job ids is handled by the pydantic model
```

Comment:
Do we still need `trans.security.encode_all_ids(rval, recursive=True)`? Are there still some models missing Pydantic conversion?

Reply:
The issue is that the fields `outputs`, `output_collections`,`implicit_collections` are created by the specified tool with regards to the input. I am not sure, if it is possible to model the created output. As such it is necessary to encode the ids here.

Reply:
What's the problem with modeling these ? Is the problem that you don't know the response shape ?

Reply:
Yes, at least for the `outputs`field. David helped me find the models for `output_collections` and `implicit_collections`.

Reply:
It should just be `Dict[str, HDA]` ... is that not so ?

Reply:
I will try that, once I got the remaining errors sorted out. Thanks!